packages/ui/src/Alert.tsx

```diff
@@ -41,12 +43,21 @@ export type AlertProps = {
   content: ReactNode
   actions?: AlertAction[]
   className?: string
-  closeToast?: (e: React.MouseEvent<HTMLButtonElement>) => void
+  closeToast?: (e?: React.MouseEvent<HTMLButtonElement>) => void
 }
 
 export const Alert: FC<AlertProps> = ({ type, title, content, actions, className, closeToast }) => {
   const { icon, ariaLabel } = ToastIcon[type]
 
+  useKey(
+    escKeyFilterPredicate,
+    (nativeEvent: KeyboardEvent) => {
+      closeToast && closeToast()
+      nativeEvent.stopImmediatePropagation()
+    },
+    { options: { once: true } },
+  )
+
```

Comment:
`useKey` uses [predicate] as  deps by default. What if `closeToast` changes at some point?

```suggestion
  const closeByEsc = useMemo(
    (nativeEvent: KeyboardEvent) => {
      closeToast?.()
      nativeEvent.stopImmediatePropagation()
    }, [closeToast]
  )
  
  useKey(
    escKeyFilterPredicate,
    closeByEsc,
    { options: { once: true } },
    [closeByEsc]
  )

```

Same goes for the Toast

Reply:
done.